libvast/test/meta_index.cpp

```diff
@@ -96,11 +96,39 @@ struct mock_partition {
 };
 
 struct fixture {
-  // Our unit-under-test.
-  meta_index meta_idx;
-
-  // Partition IDs.
-  std::vector<uuid> ids;
+  fixture() {
+    MESSAGE("generate " << num_partitions << " UUIDs for the partitions");
+    for (size_t i = 0; i < num_partitions; ++i)
+      ids.emplace_back(uuid::random());
+    std::sort(ids.begin(), ids.end());
+    // Sanity check random UUID generation.
+    for (size_t i = 0; i < num_partitions; ++i)
+      for (size_t j = 0; j < num_partitions; ++j)
+        if (i != j && ids[i] == ids[j])
+          FAIL("ID " << i << " and " << j << " are equal!");
+    MESSAGE("generate events and add events to the partition index");
+    std::vector<mock_partition> mock_partitions;
+    for (size_t i = 0; i < num_partitions; ++i) {
+      auto& part = mock_partitions.emplace_back(ids[i], i);
+      meta_idx.add(part.id, *part.slice);
```

Comment:
Having only a single slice for each partition and all slices with the same layout does not assert a lot of confidence in the correctness of the implementation.

Reply:
I was laughing because of the awesomely framed critique. 😉

Reply:
Fair enough, I had to smirk as well reading it.

Reply:
OK, I should have made myself clearer. My problem with the test "attribute extractor - &type" is that it is pretty easy to implement the functionality in a way that satisfies this test but does not work as intended. How about adding a second layout with the name "foo-plus"? That would allow you to write expressions that don't match either all or nothing.